x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -91,23 +104,29 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
           indexedStartedDate,
           libs.basePath
         );
+
         alert.scheduleActions(actionGroupId, {
-          group: '*',
+          alertDetailsUrl: getViewInAppUrl(
```

Comment:
Should we use a different function name e.g. `getAlertDetailsUrl` instead of `getViewInAppUrl`?

Reply:
Why would we want to do that?

Reply:
We have 'View in App' feature for alerts, which takes user to respective app - Metrics explorer, APM, etc. - whereas alert details is not a different app. So they are two different functionalities.

Reply:
The function that is being called is:

```
export const getViewInAppUrl = (basePath: IBasePath, relativeViewInAppUrl: string) =>
  basePath.publicBaseUrl
    ? new URL(basePath.prepend(relativeViewInAppUrl), basePath.publicBaseUrl).toString()
    : relativeViewInAppUrl;
```

The function is quite generic in its use. I could duplicate it and call it `getAlertDetailsUrl`, but I'm wondering whether thats worth it.

Alternatively I could rename the existing function and adjust the places in it. WDYT?

Reply:
@CoenWarmer I think the function is returning path like `/app/observability/alerts/alertId` and we need full url, isn't it? or I am missing something..
`getViewInAppUrl` redirects user within kibana, whereas for alert notification, user will directly click on link. So we need to modify the function or create new one.
If you use same function, then I would suggest to rename it for code clarity.

Reply:
@benakansara I assumed that the URL that is generated when developing locally does not have the basepath due to a configuration setting. In a real, deployed version, `basePath` would be filled. 

If that is not the case, then the other context variable would also be broken, because it uses the exact same function with the same variables (`basePath.publicBaseUrl`).

If you know for a fact that `basePath` should be set when running in development mode, then I agree, something is wrong.

Reply:
@CoenWarmer I tested locally with the index connector. It shows full url correctly. Not sure why it was not showing full url in email connector (screenshot in the PR description). Then the function seems to be working fine.

```
"alertDetailsLink": "http://localhost:5601/abc/app/observability/alerts/e5a073bb-642f-4f16-af9f-96dae5aa266b"
```

Reply:
Cool!

Reply:
Perhaps you have a setting in your config yaml for this that I am missing, as I still do not have the full path when devving locally.

Reply:
I have following config -
```
server.basePath: "/abc"
server.publicBaseUrl: "http://localhost:5601/abc"
```